components/camel-spring-batch/src/main/java/org/apache/camel/component/spring/batch/SpringBatchEndpoint.java

```diff
@@ -83,7 +85,7 @@ protected void doStart() throws Exception {
         if (jobLauncher == null) {
             jobLauncher = resolveJobLauncher();
         }
-        if (job == null && jobName != null) {
+        if (job == null && jobName != null && jobName.compareTo("dynamic") != 0) {
```

Comment:
please remove the ceck for "dynamic", there is nothing wrong on having a valid initial job configured. And AFAIK this concept is not used anywhere in Camel. Btw. !"dynamic".equals(jobName) is better.